### PR TITLE
Store window locations

### PR DESCRIPTION
**Pull request by lantisnt**

### CODE REVIEW

File: Modules/BiddingManager/GUI.lua [hunk 1]
```diff
@@ -51,11 +52,27 @@ end
...
 
+function BiddingManagerGUI:InitializeDB()
+    local db = MODULES.Database:GUI()
+    if not db.bidding then
+        db.bidding = { }
+    end
+    self.db = db.bidding
+end
+
+function BiddingManagerGUI:RegisterPlayerLogoutEvent()
```
Comment (SamMousa) [inline Modules/BiddingManager/GUI.lua:70]: This should be a local function; probably it shouldn't be a function at all..

Just register it in `Initialize`:
```lua
EventManager:RegisterEvent({"PLAYER_LOGOUT"}, function() StoreLocation(self) end)
```

File: Modules/PointManager/GUI.lua [hunk 2]
```diff
@@ -177,12 +194,22 @@ function PointHistoryGUI:Create()
...
     UTILS.MakeFrameCloseOnEsc(f.frame, "CLM_PointHistory_GUI")
     self.requestRefreshProfiles = true
-
     f:AddChild(CreatePointDisplay(self))
+    self:RestoreLocation()
     -- Hide by default
     f:Hide()
 end
 
+function PointHistoryGUI:StoreLocation()
```
Comment (SamMousa) [inline Modules/PointManager/GUI.lua:203]: local

File: Modules/PointManager/GUI.lua [hunk 3]
```diff
@@ -177,12 +194,22 @@ function PointHistoryGUI:Create()
...
+    self.db.location = { self.top:GetPoint() }
+end
+
+function PointHistoryGUI:RestoreLocation()
```
Comment (SamMousa) [inline Modules/PointManager/GUI.lua:207]: local

File: Modules/ProfileManager/GUI.lua [hunk 4]
```diff
@@ -32,14 +33,29 @@ local FILTER_IN_RAID = 100
...
+
+function ProfilesGUI:InitializeDB()
+    local db = MODULES.Database:GUI()
+    if not db.profiles then
+        db.profiles = { }
+    end
+    self.db = db.profiles
+end
 
+function ProfilesGUI:RegisterPlayerLogoutEvent()
```
Comment (SamMousa) [inline Modules/ProfileManager/GUI.lua:55]: local

File: Modules/ProfileManager/GUI.lua [hunk 5]
```diff
@@ -32,14 +33,29 @@ local FILTER_IN_RAID = 100
...
     self:Create()
     self:RegisterSlash()
     self._initialized = true
     LedgerManager:RegisterOnUpdate(function(lag, uncommitted)
         if lag ~= 0 or uncommitted ~= 0 then return end
         self:Refresh(true)
     end)
+end
+
+function ProfilesGUI:InitializeDB()
```
Comment (SamMousa) [inline Modules/ProfileManager/GUI.lua:47]: local

File: Modules/ProfileManager/GUI.lua [hunk 6]
```diff
@@ -332,11 +348,21 @@ function ProfilesGUI:Create()
 
     f:AddChild(CreateStandingsDisplay(self))
     f:AddChild(CreateManagementOptions(self))
-
+    self:RestoreLocation()
     -- Hide by default
     f:Hide()
 end
 
+function ProfilesGUI:StoreLocation()
```
Comment (SamMousa) [inline Modules/ProfileManager/GUI.lua:356]: local

File: Modules/ProfileManager/GUI.lua [hunk 7]
```diff
@@ -332,11 +348,21 @@ function ProfilesGUI:Create()
...
+    self.db.location = { self.top:GetPoint() }
+end
+
+function ProfilesGUI:RestoreLocation()
```
Comment (SamMousa) [inline Modules/ProfileManager/GUI.lua:360]: local

File: Modules/RaidManager/GUI.lua [hunk 8]
```diff
@@ -382,11 +399,21 @@ function RaidManagerGUI:Create()
     if ACL:IsTrusted() then
         f:AddChild(CreateManagementOptions(self))
     end
-
+    self:RestoreLocation()
     -- Hide by default
     f:Hide()
 end
 
+function RaidManagerGUI:StoreLocation()
```
Comment (SamMousa) [inline Modules/RaidManager/GUI.lua:407]: local

File: Modules/RaidManager/GUI.lua [hunk 9]
```diff
@@ -382,11 +399,21 @@ function RaidManagerGUI:Create()
...
+    self.db.location = { self.top:GetPoint() }
+end
+
+function RaidManagerGUI:RestoreLocation()
```
Comment (SamMousa) [inline Modules/RaidManager/GUI.lua:411]: local

File: Modules/Standings/GUI.lua [hunk 10]
```diff
@@ -41,6 +44,20 @@ function StandingsGUI:Initialize()
 
 end
 
+function StandingsGUI:InitializeDB()
```
Comment (SamMousa) [inline Modules/Standings/GUI.lua:47]: local

**Review by SamMousa — CHANGES_REQUESTED** [review]
All new functions should be local, they are only used internally in their respective files, as far as I can tell.

File: Modules/RaidManager/GUI.lua [hunk 11]
```diff
@@ -41,11 +42,32 @@ end
...
+    end
+end
+
 function RaidManagerGUI:Initialize()
     LOG:Trace("RaidManagerGUI:Initialize()")
     self.configuration = RosterConfiguration:New()
     self.name = ""
     self.tooltip = CreateFrame("GameTooltip", "CLMRaidListGUIDialogTooltip", UIParent, "GameTooltipTemplate")
+    InitializeDB(self)
+    EventManager:RegisterEvent({"PLAYER_LOGOUT"}, (function(...) StoreLocation(self) end))
```
Comment (SamMousa) [inline Modules/RaidManager/GUI.lua:70]: Just an FYI; in case you don't care about the parameters you are free to omit the `...` here.

**Review by SamMousa — APPROVED** [review]
LGTM, have not tested functionality in game.

**Comment by lantisnt** [comment]
Fixes also #46

**Pull request merged**